### ISSUE TITLE
Stuttering on Nvidia Shield Pro (2019) - issue with 59.940 vs 59.997 hz?

### ISSUE THREAD

**Issue by ERamseth** [moonlight-stream/moonlight-android#948]
I am experiencing stuttering on Nvidia Shield Pro (2019) only in moonlight (not in gamestream) when trying to stream in 4k at 60hz.

After looking through the nvstream logs, the two main differences are:

- Moonlight: HEVC at 59.997 FPS
- Gamestream: H264 at 59.940 FPS

Gamestream is completely smooth but Moonlight is not. 

I have tried just about everything I can think of to troubleshoot and I'm fairly sure its related to the 59.940 vs 59.997.

Things I've tried:

- setting the Shield to different output hz (59.940, 60)
- setting host PC to different output hz (59.940, 59.997, 60)
- HDR on/off
- H264 vs HEVC
- Different bandwidth settings
- Different resolutions
- Different framerate caps
- Vsync on/off
- GSync on/off
- Connected monitor to DP and HDMI
- "Never drop frames" on/off
- "Match frame rate" option in Shield menu

Equipment:
- Sony 75X850G (does 4k60 and HDR) (manually set to game mode)
- Shield Android TV Pro (2019)
- Gigabit wired ethernet
- Windows 10 PC
- RTX 3080
- Acer VG280K monitor (does 4k and HDR, various refresh rates supported and GSync works at 60hz and below... although not "certified") - connected to DP
- Latest drivers and GFE (as of 3/12/2021... not sure exact version numbers off the top of my head and I'm not at the host PC right now)

How I'm testing: Running Destiny 2 and just staying in the menu. Moving the cursor around with gamepad... in game stream it moves with zero hiccups from left to right across the screen. In Moonlight there are hiccups. Also confirmed by loading into the game itself and slowly rotating the camera. In gamestream, completely smooth. In Moonlight, stutters at seemingly regular time intervals.

**Comment by tmenzzz** [comment]
Having this exact issue on nearly identical hardware. Gamestream is smooth, but Moonlight has a consistent stutter issue.

**Comment by polson** [comment]
Also seeing microstutters on Moonlight, and none on Gamestream

**Comment by ERamseth** [comment]
This seems to have been resolved along the way somewhere. With the latest drivers, GFE, and moonlight version from play store it all just works.

**Comment by polson** [comment]
I have found that as long as the frame rate of the device that is doing the streaming matches the frame rate of the moonlight settings, and `never drop frames` is checked, then there is no skippage.

**Comment by Eitiidisuda** [comment]
I have same problem. Nvidia gamestream works great, moonlight stuttering on any bitrate settings

Equipment:

Shield Android TV 2017
Gigabit wired ethernet
Windows 10 PC
GTX 1080
Moonlight Android v9.9.3
Geforce experience 3.22.0.32

**Comment by lucius346346** [comment]
Same here.

I'm struggling with smooth playback in Moonlight for months. No matter what I can't get smooth 60 fps on Moonlight / Gamestream. I've tried everything wired/wireless LAN, changing routers/monitors, various refresh rates including creating custom ones in Nvidia panel. It happens in both android phone and Shield Pro 2019. It's generally smooth but every 1-2 second I get this single "hiccup" - It's only visible in camera panning and in UFO Test.

I've recently tried Parsec and boom - It's completely smooth. Additionally it somehow has less decoder latency 5 ms vs 20 ms. Why is it so? Wasn't gamestream suppose to be superior choice? I'd honestly prefer some fix to Moonlight/Gamestream - it's more convenient.
